File: 2586-count-the-number-of-vowel-strings-in-range/2586-count-the-number-of-vowel-strings-in-range.cpp
class Solution {
public:
    int vowelStrings(vector<string>& words, int left, int right) {
        set<char>v={'a','e','i','o','u'};
        int c=0;
        for(; left<=right; left++) {
            if((v.find(words[left][0])!=v.end())&&(v.find(words[left][words[left].length()-1])!=v.end())) c++;
        }
        return c;
    }
};

File: 2586-count-the-number-of-vowel-strings-in-range/2586-count-the-number-of-vowel-strings-in-range.rs
impl Solution {
    pub fn vowel_strings(words: Vec<String>, left: i32, right: i32) -> i32 {
        let vowels: Vec<char> = ['a', 'e', 'i', 'o', 'u'].to_vec();
        let mut c = 0;
        for i in left..right+1 {
            let mut l = false;
            let mut r = false;
            let i = i as usize;
            for char in vowels.iter() {
                if words[i].chars().nth(0).unwrap() == *char {
                    l = true;
                    break;
                }
            }
            for char in vowels.iter() {
                if words[i].chars().last().unwrap() == *char {
                    r = true;
                    break;
                }
            }
            if(l&&r) {
                c +=1;
            }
        }
        return c;

    }
}